src/validation.test.ts

```diff
@@ -617,4 +618,367 @@ describe('OpenAPIValidator', () => {
+          setMatchType: SetMatchType.Subset,
+        },
+      );
+      expect(valid.errors).toBeTruthy();
+    });
+
+    test('passes validation with any header set and setMatchType is any', async () => {
+      const valid = validator.validateResponseHeaders(
+        {
+          'X-Integer': '42',
+          'X-String': 42,
+        },
+        'listPets',
+        {
+          statusCode: 200,
+          setMatchType: SetMatchType.Any,
+        },
+      );
+      expect(valid.errors).toBeFalsy();
+    });
+
+    test('fails validation with an invalid response object', async () => {
+      const valid = validator.validateResponseHeaders(
+        {
+          unknown: 'property',
+        },
+        'listPets',
+        {
+          statusCode: 200,
+          setMatchType: SetMatchType.Exact,
+        },
+      );
+      expect(valid.errors).toBeTruthy();
+    });
+
+    test('fails validation with a missing response object', async () => {
+      const valid = validator.validateResponseHeaders(
+        null,
+        'listPets',
+        {
+          statusCode: 200,
+          setMatchType: SetMatchType.Exact,
+        },
+      );
+      expect(valid.errors).toBeTruthy();
+    });
+
+    test('passes validation with a res code handler fallback', async () => {
+      const valid = validator.validateResponseHeaders(
+        {
+          'X-Other-Integer': '42',
+          'X-Other-String': 42,
+          'X-Other-Boolean': 'true',
+        },
+        'listPets',
+        {
+          statusCode: 205,
+          setMatchType: SetMatchType.Exact,
+        },
+      );
+      expect(valid.errors).toBeFalsy();
+    });
+
+    test('throw an error when passing SetMatchType', async () => {
+      expect(() => {
+        validator.validateResponseHeaders(
+          {
+            'X-Other-Integer': '42',
+            'X-Other-String': 42,
+            'X-Other-Boolean': 'true',
+          },
+          'listPets',
+          {
+            statusCode: 205,
+            setMatchType: 'error' as SetMatchType,
+          },
+        );
+      }).toThrow();
+    });
+
+    test('passes validation with header case not matching the spec', async () => {
+      const valid = validator.validateResponseHeaders(
+        {
+          'X-OtHer-InTeger': '42',
+          'x-other-string': 42,
+          'X-OTHER-BOOLEAN': 'true',
+        },
+        'listPets',
+        {
+          statusCode: 205,
+          setMatchType: SetMatchType.Exact,
+        },
+      );
+      expect(valid.errors).toBeFalsy();
+    });
+
+    test('fails validation with header separators omitted', async () => {
+      const valid = validator.validateResponseHeaders(
+        {
+          'xotherinteger': '42',
+          'xotherstring': 42,
+          'XOTHERBOOLEAN': 'true',
+        },
+        'listPets',
+        {
+          statusCode: 205,
+          setMatchType: SetMatchType.Exact,
+        },
+      );
+      expect(valid.errors).toBeTruthy();
+    });
+  });
```

Comment:
I feel like it would be good to add one test with only the two required params for `validateResponseHeaders`. Right now, all tests include the 3rd `opts` param.

Reply:
Good idea, thanks.

Reply:
Good idea, thanks.